src/main/java/io/temporal/samples/hello/HelloPeriodic.java

```diff
@@ -200,11 +242,40 @@ public static void main(String[] args) throws InterruptedException {
                 .setTaskQueue(TASK_QUEUE)
                 .build());
 
+    // Execute our workflow.
     try {
-      // execute our workflow
       WorkflowClient.start(workflow::greetPeriodically, "World");
+      System.out.println("Started a new GreetingWorkflow.");
     } catch (WorkflowExecutionAlreadyStarted e) {
-      System.out.println("Started: " + e.getMessage());
+      // If the workflow is already running, we cannot start it. Instead, we will connect to the
+      // existing instance.
+      workflow = client.newWorkflowStub(GreetingWorkflow.class, WORKFLOW_ID);
+      System.out.println("Connected to an existing GreetingWorkflow.");
+    }
+
+    // The workflow is running now. In this example, we pause for a few seconds to observe its output.
+    final int ObservationPeriodSecs = 20;
+    System.out.println("Observing the workflow execution for " + ObservationPeriodSecs + " seconds.");
+
+    try {
+      Thread.sleep(ObservationPeriodSecs * 1000);
+    } catch (InterruptedException intEx) {
+      intEx.printStackTrace();
     }
+
+    // Signal the workflow to exit.
+    System.out.println("Requesting the workflow to exit.");
+    workflow.requestExit();
+
+    // In real life we could exit now. However, in this example the workflow is running in the
```

Comment:
this whole comment is confusing,

pls just say "allow workflow to finish before the worker is shut down." 

that is really the reason why workflow is still running if you exist right away, and not "to allow it to finish gracefully" 
if user started this sample back up the worker would pick up the wf task and finish it.

Reply:
✔️